test/integration/kube/nnc.go

```diff
@@ -0,0 +1,22 @@
+package kube
+
+import (
+	"math"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/crd/nodenetworkconfig/api/v1alpha"
+	"github.com/stretchr/testify/assert"
+)
+
+func ValidateNNC(t *testing.T, nnc v1alpha.NodeNetworkConfig, pods int64, subnetScarcity bool) {
```

Comment:
The only place this is being validated is where subnet scarcity is set to false in the k8s_test. Should we not be validating the NNC after scaling up the pods such that subnet scarcity is true and then again after scaling down when subnet scarcity is false again?